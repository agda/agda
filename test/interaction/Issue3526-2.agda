{-# OPTIONS --prop #-}

module Issue3526-2 where

open import Agda.Builtin.Equality

record Truth (P : Prop) : Set where
  constructor [_]
  field
    truth : P
open Truth public

data ⊥' : Prop where

⊥ = Truth ⊥'

¬ : Set → Set
¬ A = A → ⊥

unique : {A : Set} (x y : ¬ A) → x ≡ y
unique x y = refl

⊥-elim : (A : Set) → ⊥ → A
⊥-elim A b = {!!}

open import Agda.Builtin.Bool
open import Agda.Builtin.Unit

set : Bool → Set
set false = ⊥
set true  = ⊤

set-elim : ∀ b → set b → Set
set-elim b x = {!!}

