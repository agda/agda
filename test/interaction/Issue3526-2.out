(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : A ?1 : Set " nil)
((last . 1) . (agda2-goals-action '(0 1)))
((last . 2) . (agda2-make-case-action '("‚ä•-elim A ()")))
((last . 1) . (agda2-goals-action '(0 1)))
((last . 2) . (agda2-make-case-action '("set-elim true x = ?")))
((last . 1) . (agda2-goals-action '(0 1)))
